python_modules/dagster-ext/dagster_ext/__init__.py

```diff
@@ -543,6 +543,80 @@ def upload_messages_chunk(self, payload: IO, index: int) -> None:
 # ########################
 
 
+class IContext(ABC):
```

Comment:
What should the context be named? `IContext` is a place holder for now. Some options:
* `AbstractExecutionContext` - potentially too similar to [`AbstractComputeExecutionContext`](https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/execution/context/compute.py#L53)
*

Reply:
Can't think of a good one but I'm pretty opposed to `IExecutionContext` because `OpExecutionContext` doesn't implement this, so it would be very confusing.

Maybe `INewExecutionContext` or something (this isn't user-facing right)? `IMaterializationContext`?

Might help to spell out some of the thinking behind having a common interface here? What is it that `AssetExecutionContext` and `ExtContext` share in common? That they both proxy to an `OpExecutionContext`?

Reply:
I saw feedback from dogfooding that we should ditch the term "context" entirely which could be useful to avoid confusing with our orchestration process context APIs. We could call it an `ExtRequest` or `ExtInvocation`

Reply:
the resulting class signature for `AssetExecutionContext` would be a bit weird then 

```python
class AssetExecutionContext(OpExecutionContext, ExtRequest)

# and eventually 
class AssetExecutionContext(ExtRequest)
```
it wouldn't be super visible to users, so it's probably ok, but just looks a touch odd to me

Reply:
@schrockn was that feedback concerning the orchestration-side `ExtOrchestrationContext` or ext-side `ExtContext`? The above conversation is about an interface implemented by `ExtContext` but this feedback seems more fitting for `ExtOrchestrationContext`.

Reply:
@jamiedemaria I agree that `class AssetExecutionContext(ExtRequest)` is super goofy and we should not have that in the codebase.

I bias towards someting like a `IHasUserFacingExecutionMethods` marker inteface. I think this is better than `IHasCoreAssetMethods` because the `ExtContext` interface can be used right now for vanilla ops. Not exporting it so that we have the optional value to change it, making a "two-way door"/reversible decision, so changing our mind has minimal cost.

This is for our only bookkeeping only, and we should feel free to change the name as we get more clarity around what it represents.  `IHasUserFacingExecutionMethods` is long and goofy, but it is very specific, and I think _accurately_ communicates that this (our user-facing context API, speciflcally) isn't the most conceptually coherent part of the system at the moment, and are just stopping the bleeding from adding yet another similar API. The only purpose of the interface is to keep ourselves honest.